packages/astro/src/runtime/server/index.ts

```diff
@@ -26,24 +29,24 @@ const svgEnumAttributes = /^(autoReverse|externalResourcesRequired|focusable|pre
 // Or maybe type UserValue = any; ?
 async function _render(child: any): Promise<any> {
 	child = await child;
-	if (child instanceof UnescapedString) {
+	if (child instanceof HTMLString) {
 		return child;
 	} else if (Array.isArray(child)) {
-		return unescapeHTML((await Promise.all(child.map((value) => _render(value)))).join(''));
+		return markHTMLString((await Promise.all(child.map((value) => _render(value)))).join(''));
 	} else if (typeof child === 'function') {
 		// Special: If a child is a function, call it automatically.
 		// This lets you do {() => ...} without the extra boilerplate
 		// of wrapping it in a function and calling it.
 		return _render(child());
 	} else if (typeof child === 'string') {
-		return escapeHTML(child);
+		return markHTMLString(escapeHTML(child));
```

Comment:
This is the main fix, the rest of this PR is mainly renaming: once it's been escaped, we can mark it as a valid HTMLString. This prevents it from being escaped again.